### ISSUE TITLE
FF sending collection name as Index name to Algolia

### ISSUE THREAD

**Issue by laughingbuddha24** [FlutterFlow/flutterflow-issues#2111]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

FF is not pulling index data from Algolia, I have confirmed that all keys and configuration settings are correct, and that collection data is exporting from Firestore to the Algolia index, and I still get a loading icon that never resolves.

![results-page](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/c72809d1-2b22-4c99-8d24-d06fa987b3ab)

In the browser debug console I see an error from Algolia stating 'Index test_collection does not exist, status: 404'. 

![console-error](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/8c8f9136-9969-41aa-b58c-1aa13d276df3)

It appears that FF is requesting the collection name from Algolia rather than the index name. Images below show that the index and collection names have been set correctly in Firebase and Algolia.

![index-name](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/8703731a-0d9c-409e-a402-6a52ebcf82f4)

![firestore-algolia-config](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/3bfb6441-9905-4055-be2e-af223c476d3d)

### Expected Behavior

The Listview widget in the test page UI should show all documents in the collection.

### Steps to Reproduce

1. create a new page
2. add widgets as per screenshot below and configure query and variables as per 
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/e8a5a58a-d202-4719-90e8-bd37aef32b1d)
3. load a test session

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4Skcrq2ItNvttZ+KXqdcAxnSQWOUA5aLgriOtqEBI3faKvELNyXOjASEhVXd+/TAl2Ilv8mnk03M2PveOeCOQHAweBc4s51pZySTzMYE+uRcW4BqiwVUdQBeVUJEuVyJ6Jnx17HMhoV14hwV2yOe6taC3fGZTtOTANPP+hIdnLpHDlAi+6Xn0Jg1ZSfCvr

### Context

I'm making a collection searchable by app users so they can select from a list of gyms in the area.

### Visual documentation

See images in description above

### Environment

```markdown
- FlutterFlow version: 3.13.7
- Platform: Web
- Browser name and version: Chrome 120.0.6099.130 
- Operating system and version affected: Windows 11 Home version 22H2
```

**Comment by hariprasadms** [comment]
Hi @laughingbuddha24 - Thank you for submitting the issue. I have reviewed your project, and the configurations related to Algolia seem to be correct. However, as I do not have access to your Firebase project, I am unable to investigate potential issues with Algolia and Firebase configurations. In my test projects as per my latest tests Algolia works without any errors.
In this case, I recommend testing the integration with more than one collection to assess the consistency of the issue. If the problem persists, it would be advisable to reach out to the support team to have your project thoroughly checked. At this stage, it appears to be a possible misconfiguration on your end. Please carefully check and confirm back to me with details regarding the consistency of the issue.

**Comment by laughingbuddha24** [comment]
Hi @hariprasadms, I have tried it on my app under development and on a blank test app and I get the same result. The full configuration is below - is anything missing? The only detail not shown here is the Algolia API key but I have entered that correctly in Firebase.

![algolia-1](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/4fcf4d5d-5abb-453d-a240-a18f8bfc22e3)

![firebase-1](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/342141eb-d8ff-4478-b79d-5876b7f7d8f5)

![ff-1](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/092cd033-c979-456c-8621-d0fda68c05b6)

![ff-2](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/f9654c68-c22d-4bba-a11a-5f973b3cb13c)

![ff-3](https://github.com/FlutterFlow/flutterflow-issues/assets/155855214/94b16d21-a7b8-45c8-9ae5-959d7a8937e3)

**Comment by hariprasadms** [comment]
Thanks @sbis04.
@laughingbuddha24 - I hope you found the solution now. If problem no more exists, please close the issue. Thanks

**Comment by laughingbuddha24** [comment]
You're right it does say that in the documentation. I read that many times and understood it to mean 'go to the Algolia dashboard and create a new index (of any name)'. 'Enter the same name in this field' I took to mean enter the name you just created in the Firestore index name field. If you change the second sentence to 'Enter the collection name in this field' I'm sure it will avoid confusion in the future

Thanks for following up, I appreciate it.

**Comment by laughingbuddha24** [comment]
Hi Souvik,

Thanks for following up I appreciate it. It was a confusing sentence in the documentation that tripped me up. See my response in Github if you want to update one sentence in the documentation. I found some people in reddit with the same issue so maybe it will avoid confusion in the future.

Kind regards,

Dom O'Brien
Laughing Buddha Design
Mobile: [0432 84 22 00](tel:0432208420222000)
[www.laughingbuddhadesign.com​](https://www.laughingbuddhadesign.com)
***@***.******@***.***%E2%80%8B)
